custom_metrics/almanac.js

```diff
@@ -20,21 +20,26 @@ function getNodeAttributes(node) {
   return node.cloneNode(false).attributes;
 }
 
+function parseNode(node) {
+  var attributes = Object.values(getNodeAttributes(node));
+  var el = {};
+
+  el.tagName = node.tagName.toLowerCase(); // for reference
+  for (var n = 0, len2 = attributes.length; n < len2; n++) {
+    var attribute = attributes[n];
+    el[attribute.name.toLowerCase()] = attribute.value;
```

Comment:
From checking exceptions in last years run I found that attribute.name could sometimes be missing. This check fixed it:

```
if (attribute.name) {
            el[attribute.name.toLowerCase()] = attribute.value;
}
```

Reply:
Where did you see that? I don't think this caused any problems in last years run

Reply:
It did. I queried the 2000 examples table and it identified 31 pages where the amanac property was null (1.5%). I then tested a few of them and they caused a null exception on attribute.name. The one I first found the error on seemed to have an attribute called "3" with no value dynamically added to a node?

Reply:
I'll go ahead and add it in then. Great catch!